src/runtime/dict.cpp

```diff
@@ -559,8 +561,18 @@ Box* dictNonzero(BoxedDict* self) {
 
 Box* dictFromkeys(Box* cls, Box* iterable, Box* default_value) {
     auto rtn = new BoxedDict();
-    for (Box* e : iterable->pyElements()) {
-        dictSetitem(rtn, e, default_value);
+    if (PyAnySet_Check(iterable)) {
+        for (auto&& elt : ((BoxedSet*)iterable)->s) {
+            auto it = rtn->d.find(elt);
+            if (it != rtn->d.end())
+                it->second = default_value;
```

Comment:
This looks a bit odd -- I'm not sure that updating the iterator is guaranteed to mutate the underlying container.  I tried to test it but 1) it looks like we don't have any test coverage of this line of code :)  and 2) if the element already exists, shouldn't it already be set to default_value?

Maybe we could skip the find() step and just do an insert().